src/Akka.Streams.Kafka/Helpers/Committer.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Threading.Tasks;
+using Akka.Annotations;
+using Akka.Streams.Dsl;
+using Akka.Streams.Kafka.Messages;
+using Akka.Streams.Kafka.Settings;
+
+namespace Akka.Streams.Kafka.Helpers
+{
+    /// <summary>
+    /// Implements committing flows
+    /// </summary>
+    public static class Committer
+    {
+        /// <summary>
+        /// Batches offsets and commits them to Kafka, emits <see cref="CommittableOffsetBatch"/> for every committed batch.
+        /// </summary>
+        public static Flow<ICommittable, ICommittableOffsetBatch, NotUsed> BatchFlow(CommitterSettings settings)
+        {
+            return Akka.Streams.Dsl.Flow.Create<ICommittable>().GroupedWithin(settings.MaxBatch, settings.MaxInterval)
+                .Select(CommittableOffsetBatch.Create)
+                .SelectAsync(settings.Parallelism, async batch =>
+                {
+                   await batch.Commit();
+                   return batch;
+                });
+        }
+
+        /// <summary>
+        /// Batches offsets and commits them to Kafka, emits <see cref="Done.Instance"/> for every committed batch.
+        /// </summary>
+        public static Flow<ICommittable, Done, NotUsed> Flow(CommitterSettings settings)
+        {
+            return BatchFlow(settings).Select(_ => Done.Instance);
+        }
+
+        /// <summary>
+        /// API MAY CHANGE
+        /// 
+        /// Batches offsets from context and commits them to Kafka, emits no useful value,
+        /// but keeps the committed <see cref="ICommittableOffsetBatch"/> as context
+        /// </summary>
+        [ApiMayChange]
+        public static FlowWithContext<ICommittableOffset, E, ICommittableOffsetBatch, NotUsed, NotUsed> FlowWithOffsetContext<E>(CommitterSettings settings)
+        {
+            var value = Akka.Streams.Dsl.Flow.Create<Tuple<E, ICommittableOffset>>()
```

Comment:
@Aaronontheweb I had to use `Tuple<K,V>` here because `FlowWithContext` class from Akka.Streams beta2 still uses it instead of `ValueTuple`. I remember that you wanted to move from `Tuple` class in new Akka.NET version, so is it something you are going to change in the future?

Reply:
Yeah, I need to update that in Akka.Streams